src/Storage/Storage/Common/Cmdlet/SetAzureStorageServiceMetrics.cs

```diff
@@ -216,50 +266,75 @@ public override void ExecuteCmdlet()
             else //Table use old XSCL
             {
                 StorageTableManagement tableChannel = new StorageTableManagement(Channel.StorageContext);
-                XTable.ServiceProperties currentServiceProperties = tableChannel.GetStorageTableServiceProperties(GetTableRequestOptions(), TableOperationContext);
 
-                // Premium Account not support classic metrics and logging
-                if ((MetricsType == ServiceMetricsType.Hour && currentServiceProperties.HourMetrics == null)
-                    || (MetricsType == ServiceMetricsType.Minute && currentServiceProperties.MinuteMetrics == null))
+                if (tableChannel.IsTokenCredential)
                 {
-                    AccountProperties accountProperties = Channel.GetAccountProperties();
-                    if (accountProperties.SkuName.Contains("Premium"))
```

Comment:
Why remove the Premium account related code? The error should only be throw on premium account.